src/lib/utils/longRunningProcessTimer.ts

```diff
@@ -0,0 +1,121 @@
+import {PerformanceObserver, performance} from 'perf_hooks';
+import {red} from 'colors';
+const utils = require('./utils.js');
+import prettyMs from 'pretty-ms';
+import { Logger } from '../../../src/typings/logger';
+const ora = require('ora');
+
+export default class LongRunningProcessTimer{
+  private _logger: Logger;
+  private _packageName: string;
+  private _version: string;
+  private _showSpinner: boolean;
+  private _spinnerStyle: string;
+  private _interval: number;
+  private _processStartingMsg: string;
+  private _processOngoingMsg: string;
+  private _processFinishedMsg: string;
+  private _longRunningThreshold: number;
+  private _startMark: string;
+  private _ongoingMark: string;
+  private _downloadOngoing: string;
+  private _endMark: string;
+  private _downloadComplete: string;
+  private _observer!: PerformanceObserver;
+  private _spinner!: any;
+  private _intOngoingDownload!: NodeJS.Timeout;
+
+  constructor(logger: Logger, packageName: string, version: string, processStartingMsg: string, processOngoingMsg: string, processFinishedMsg: string, showSpinner?:boolean, spinnerStyle?: string, interval?: number, longRunningThreshold?: number){
+    this._logger = (logger && typeof logger.info === 'function') ? logger : console;
```

Comment:
Is there any place where logger is not the logger object?

By adding type we should not need to have those check

Reply:
true, good catch!